### DISCUSSION TITLE
[2023-05-15] GUI Demo Day

### DISCUSSION THREAD

**Discussion by vitvakatu** [enso-org/enso#6698] [Weekly Demo Day]
Last week I fixed a bug with an invisible input port on some nodes: https://github.com/enso-org/enso/issues/6485. (see the video)

And also spent quite some time debugging WebGL context loss in the PR: https://github.com/enso-org/enso/pull/6321. I didn't find the exact reason for this bug, but it turned out that it was accidentally fixed at some point in develop. While investigating, I added simple code to debug async tasks in our source code. I think it can become helpful in the future. (I'm cleaning up this code at the moment)

https://user-images.githubusercontent.com/6566674/237079531-23934966-8f72-4675-abe3-78a3f0c0cda4.mp4

**Comment by wdanilo** [comment]
Things on my side:
- Finished [Dynamic app resolution resampling](https://github.com/enso-org/enso/pull/6595)
- Discovered why searching in component browser is so slow. It turns out, that on weaker hardware the majority of the time is occupied by style-recalculation and re-layouting of DOM elements. We are planning fixes to that right now.
   ![image](https://github.com/enso-org/enso/assets/1623053/294a0e94-04f0-4272-9151-34db1075b917)
- Discovered other issues. On fast computers, the above mentioned slowdown takes 50% of the time, where another 50% is taken probably by our not-performant FRP implementation. I started working on a new FRP implementation using a drastically different approach, inspired by talks with @Frizi and @kazcw. This new implementation should be done till the end of the week as in-place replacement (if everything goes well). This should drastically improve FRP initialization time and also sending things trough FRP, as it will not clone things everywhere, it will pass them by reference instead.

**Comment by MichaelMauderer** [comment]
Fix FRP events that deactivate visualizations. #6638 

https://github.com/enso-org/enso/assets/1428930/6e324ddf-f365-48b8-bb2a-c68b2fbd24ef

https://github.com/enso-org/enso/assets/1428930/437f7822-7c35-48ba-a055-59d6f712a813

(more in progress)

**Comment by farmaazon** [comment]
I spent my 4 days of working mostly [failing](https://github.com/enso-org/enso/issues/6556#issuecomment-1541877957) [to](https://github.com/enso-org/enso/issues/6556#issuecomment-1543637356) [reproduce](https://github.com/enso-org/enso/issues/6579#issuecomment-1545479742) bugs. Closed some of them.

Fixed the code representing file still being uploaded. Now it refers to actual method in the Stdlib, making the proper "File uploading" error instead of compile error.

![image](https://github.com/enso-org/enso/assets/3919101/75d69025-b978-4eb8-9f28-fef766e7e2b2)

Also, temporarily skipped checking the checksum of uploaded file, because used method of the engine returns error for big files, what [was reported](https://github.com/enso-org/enso/issues/6691).

**Comment by mwu-tow** [comment]
Mostly I've been working on debugging Undo/Redo issues; this is still in progress. https://github.com/enso-org/enso/issues/6435
Other than that, some CI tweaks (nightlies every day) were done but nothing major.

**Comment by Procrat** [comment]
- #6617
  When selecting a nested breadcrumb, a race condition would occur and the breadcrumbs sometimes switched positions.

  Before:

  https://user-images.githubusercontent.com/1428930/234867978-1549bf40-4dea-4a68-8a24-d3cc6228b833.mp4
  
  After:

  https://github.com/enso-org/enso/assets/607786/73734805-292d-42ce-b08a-060a987d99bb

- #6620
  The `ctrl-r` shortcut used to reset the project to the last savepoint. Since this shortcut is easily accessible but can potentially throw away lots of work, it got changed to `ctrl-shift-r`.
  As a consequence, the program execution shortcuts changed from <kbd>cmd</kbd>+<kbd>shift</kbd>+<kbd>t</kbd>/<kbd>r</kbd> to <kbd>cmd</kbd>+<kbd>alt</kbd>+<kbd>t</kbd>/<kbd>r</kbd>.
  Follow-up issues were made to make this more user-friendly:
  - #6624 
  - #6625

- #6663
  The shortcut to open the full-screen visualisation is now `shift-space` so it doesn't interfere with the `space` shortcut to toggle the mini-visualisation.

**Comment by Frizi** [comment]
- continuing effort towards smooth list editor experience - many more bugs fixed and waiting for third QA round to complete - https://github.com/enso-org/enso/pull/6540
- Did some progress on new node design, for now decided to ignore issues with color assignment and focus on the general design first. Added an option to disable precompiled shaders in IDE and fixed old interactive mode stylesheet editing in devtools.
<img width="238" alt="image" src="https://github.com/enso-org/enso/assets/919491/86ee4a82-2ae0-4220-818d-697a39b90181">
<img width="143" alt="image" src="https://github.com/enso-org/enso/assets/919491/5f451b5b-4c2a-4032-ab3a-4ae7b641909c">
